ui/src/app/machines/components/HeaderStrip/ActionFormWrapper/DeployForm/DeployFormFields/DeployFormFields.js

```diff
@@ -25,6 +26,15 @@ export const DeployFormFields = () => {
   const canBeKVMHost =
     values.oSystem === "ubuntu" && values.release === "bionic";
 
+  // Whenever the selected OS changes, set the release value to the first OS's
+  // release in state.
+  const previousOS = usePrevious(values.oSystem);
+  useEffect(() => {
```

Comment:
I can't quite figure out why this is in a `useEffect` and not on the field's `onChange`. Is there a reason? If so, maybe a comment would be useful here for future developers (probably me) wondering the same thing.

Reply:
Basically because the list of releaseOptions is determined by the hook:

``` js
const releaseOptions = useSelector((state) =>
  generalSelectors.osInfo.getOsReleases(state, values.oSystem)
);
```
And since you can't have hooks in callbacks I've moved the logic up into the main component. I guess I could just have get all releaseOptions and then filter based on values.oSystem *outside* of the hook instead though.